nautobot/utilities/forms/fields.py

```diff
@@ -697,23 +697,24 @@ def _check_values(self, values):
                 code="invalid_list",
             )
         pk_values = set()
+        natural_key_values = set()
         for item in values:
             query = Q()
             try:
                 # query pk field if this is a uuid object or a valid uuid string
                 isinstance(item, uuid.UUID) or uuid.UUID(item)
                 pk_values.add(item)
                 query |= Q(pk=item)
-            except (ValueError, TypeError):
-                pass
-            query |= Q(**{self.natural_key: str(item)})
+            except (ValueError, TypeError, AttributeError):
+                natural_key_values.add(item)
+                query |= Q(**{self.natural_key: str(item)})
```

Comment:
I wonder - should we avoid casting `item` to `str` in this case?

The one thing we potentially miss out on, I think, by moving this line inside the `except` block, is that if an object has a natural key that is (or looks like) a UUID, we won't get this exception and so the value won't be added to the natural key values. Not sure how critical that is, though.

Reply:
I'm not sure there is currently any model field that is/looks like a UUID but not a `pk`/`id`.